dart/lib/src/sentry_client.dart

```diff
@@ -22,7 +22,13 @@ abstract class SentryClient {
   SentryClient.base(this._options, {String origin}) {
     _random = _options.sampleRate == null ? null : Random();
     if (_options.transport is NoOpTransport) {
-      _options.transport = HttpTransport(options: _options, origin: origin);
```

Comment:
@marandaneto  shouldn't we catch the potential Dns.parse exception ? What do you think would be the best strategy or preferred wording here ?

Reply:
on Java we do on Sentry.init

https://github.com/getsentry/sentry-java/blob/4a9b3cc961e74421b8c56b77cc4808c5ddae2d5b/sentry/src/main/java/io/sentry/Sentry.java#L181-L190

ideally, we try to parse on https://github.com/getsentry/sentry-dart/blob/feature/unified-api/dart/lib/src/sentry.dart#L112-L121

so we crash on Sentry.init and people can fix right away